RSyntaxTextArea/src/main/java/org/fife/ui/rsyntaxtextarea/modes/LatexTokenMaker.flex

```diff
@@ -65,66 +36,51 @@ import org.fife.ui.rsyntaxtextarea.*;
 
 
 	/**
-	 * Constructor.  This must be here because JFlex does not generate a
+	 * Constructor. This must be here because JFlex does not generate a
 	 * no-parameter constructor.
 	 */
 	public LatexTokenMaker() {
 	}
 
+    @Override
+    public void addNullToken() {
+        if(currentState >= 0) {
+            super.addNullToken();
+        } else {
+            super.addToken(s.array, start, start - 1, currentState, start);
+        }
+    }
 
-	/**
-	 * Adds the token specified to the current linked list of tokens.
-	 *
-	 * @param tokenType The token's type.
-	 * @see #addToken(int, int, int)
-	 */
-	private void addHyperlinkToken(int start, int end, int tokenType) {
-		int so = start + offsetShift;
-		addToken(zzBuffer, start,end, tokenType, so, true);
-	}
-
+    protected void addNullToken(int value) {
+        super.addToken(s.array, start, start - 1, value, start);
+    }
 
-	/**
+    /**
 	 * Adds the token specified to the current linked list of tokens.
 	 *
 	 * @param tokenType The token's type.
 	 */
 	private void addToken(int tokenType) {
-		addToken(zzStartRead, zzMarkedPos-1, tokenType);
-	}
-
-
-	/**
-	 * Adds the token specified to the current linked list of tokens.
-	 *
-	 * @param tokenType The token's type.
-	 * @see #addHyperlinkToken(int, int, int)
-	 */
-	private void addToken(int start, int end, int tokenType) {
-		int so = start + offsetShift;
-		addToken(zzBuffer, start,end, tokenType, so, false);
+        // take care of the math mode
+        if(currentState != 0) {
+            super.setLanguageIndex(1);
+        } else {
+            super.setLanguageIndex(0);
+        }
+
+		super.addToken(s.array, s.offset + zzStartRead, s.offset + zzMarkedPos - 1, tokenType, start + zzStartRead);
 	}
 
+    private void addToken(int tokenType, int languageIndex) {
+        super.setLanguageIndex(languageIndex);
+        super.addToken(s.array, s.offset + zzStartRead, s.offset + zzMarkedPos - 1, tokenType, start + zzStartRead);
+    }
 
-	/**
-	 * Adds the token specified to the current linked list of tokens.
-	 *
-	 * @param array The character array.
-	 * @param start The starting offset in the array.
-	 * @param end The ending offset in the array.
-	 * @param tokenType The token's type.
-	 * @param startOffset The offset in the document at which this token
-	 *                    occurs.
-	 * @param hyperlink Whether this token is a hyperlink.
-	 */
-	@Override
-	public void addToken(char[] array, int start, int end, int tokenType,
-						int startOffset, boolean hyperlink) {
-		super.addToken(array, start,end, tokenType, startOffset, hyperlink);
-		zzStartRead = zzMarkedPos;
+    private void addHyperlinkToken(int tokenType) {
+        super.setLanguageIndex(0); // Hyperlinks cannot be found into math blocks
+		super.addToken(s.array, s.offset + zzStartRead, s.offset + zzMarkedPos - 1, tokenType, start + zzStartRead, true);
 	}
 
-
 	/**
 	 * ${inheritDoc}
 	 */
```

Comment:
Can you remove this unnecessary `inheritDoc` block comment as well? Not related to your PR, but just to tidy things up a bit